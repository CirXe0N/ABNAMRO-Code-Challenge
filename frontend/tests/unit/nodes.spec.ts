import { mount } from '@vue/test-utils';
import NodesComponent from '@/components/Nodes.vue';
import fetchMock from 'jest-fetch-mock';

fetchMock.enableMocks();
let nodes: any[];

describe('Nodes.vue', () => {
  beforeEach(() => {
    nodes = [{
      _id: 38,
      _labels: [
        'Node',
      ],
      name: 'A',
      description: 'This is a description of A',
    },
    {
      _id: 37,
      _labels: [
        'Node',
      ],
      name: 'B',
      description: 'This is a description of B',
      parent: {
        _id: 12,
        _type: 'IS_PARENT_OF',
        node: {
          _id: 38,
          _labels: [
            'Node',
          ],
          name: 'A',
          description: 'This is a description of A',
        },
      },
    }];
  });

  it('renders the nodes', () => {
    fetchMock.mockResponseOnce(JSON.stringify(nodes), { status: 200 });
    const wrapper = mount(NodesComponent);
    (wrapper.vm as any).generateGraph();
    expect(wrapper.element).toMatchSnapshot();
  });
});
