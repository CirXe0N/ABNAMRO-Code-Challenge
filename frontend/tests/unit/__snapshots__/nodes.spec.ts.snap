// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nodes.vue renders the nodes 1`] = `
<div
  id="nodes-graph"
>
  <div
    id="controls"
  >
    <div>
      Use the mouse wheel to 
      <span>
        zoom
      </span>
       in/out.
    </div>
     
    <div>
      Click and drag to 
      <span>
        pan
      </span>
       canvas.
    </div>
  </div>
   
  <div
    class="dialog"
  >
    <div
      class="dialog__overlay"
    />
     
    <div
      class="dialog__content"
    >
      <div
        class="morph-shape"
      >
        <svg
          height="100%"
          preserveAspectRatio="none"
          viewBox="0 0 560 280"
          width="100%"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            fill="none"
            height="276"
            width="556"
            x="3"
            y="3"
          />
        </svg>
      </div>
       
      <div
        class="dialog-inner"
      >
        <h2>
          Node: 
        </h2>
         
        <p>
          
        </p>
         
        <button>
          Close
        </button>
      </div>
    </div>
  </div>
</div>
`;
